### PR TITLE
ACENAS_CARREON_LAPITAN_SANJUAN Hyperparameter Tuning Function and Model Architecture + Concerns about Timestamp Feature

### PR DESCRIPTION
**Pull request by FreyaKassie16**
### For Review
Updated model architecture to take inputs from the 4 datasets (phone_accel, phone_gyro, watch_accel, watch_gyro). Hyperparameters to be tuned are number of hidden layers, number of neurons per hidden layer, learning rate, optimizer and their parameters, and batch size. We would appreciate your comments and any recommendations for the model architecture as can be seen in the build_model() function.

### Concerns
The timestamp in each of the 4 datasets, which appears to be in UNIX time format, seem to be full of errors. As can be seen below from the phone_accel dataset for example, the timestamps have a wide range of values. When converted to a datetime format, the date can range from July 1, 1978 to September 1, 2027 (Yes, 2027, not a typo). This problem is also present in the 3 other datasets. How should we proceed with handling this feature?
Screenshot 2024-04-17 191056

### CODE REVIEW

**Comment by FreyaKassie16** [comment]
UNIX was indeed in nanoseconds, used pd.to_datetime which has units up to nanoseconds to convert it.

**Comment by FreyaKassie16** [comment]
Finalized notebook, ready for submission

**Comment by FreyaKassie16** [comment]
Model Architecture:
model

Results:
image

**Comment by FreyaKassie16** [comment]
> Feature Engineering needs review in regards to resampling of time series data.

For the concern on the resampling of time series data due to a a lack of data (0) for many consecutive time periods in the visualization of x, y, and z over time, the phone accelerometer data was sorted by the timestamp column from earliest to latest, and differencing was applied to that column. The code and results can be seen below. There are consecutive observations that are hours, and even days, apart from each other. 

image
